File: solutions/kotlin/hamming/1/src/main/kotlin/Hamming.kt
object Hamming {

    fun compute(leftStrand: String, rightStrand: String): Int {
        if (leftStrand.length != rightStrand.length) throw IllegalArgumentException("left and right strands must be of equal length")

        return leftStrand.zip(rightStrand)
            .filterNot { it.first == it.second }
            .map { 1 }
            .sum()
    }
}

File: solutions/scala/hamming/1/src/main/scala/Hamming.scala
object Hamming{
    def distance(seq1: String, seq2: String): Option[Int] = {
        if (seq1.length != seq2.length) return None
        val zipped = seq1.toCharArray.zip(seq2.toCharArray())
        Some(zipped.map(x => if (x._1 == x._2) 0 else 1).sum)
    }
}